services/alertmanager/alertmanager.go

```diff
@@ -743,77 +741,128 @@ func (svc *Service) GetAlerts(ctx context.Context) ([]*ammodels.GettableAlert, e
 	return resp.Payload, nil
 }
 
-// FindAlertByID searches alert by ID in alertmanager.
-func (svc *Service) FindAlertByID(ctx context.Context, id string) (*ammodels.GettableAlert, error) {
+// FindAlertsByID searches alerts by IDs in alertmanager.
+func (svc *Service) FindAlertsByID(ctx context.Context, ids []string) ([]*ammodels.GettableAlert, error) {
 	alerts, err := svc.GetAlerts(ctx)
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to get alerts form alertmanager")
 	}
 
+	m := make(map[string]struct{}, len(ids))
+	for _, id := range ids {
+		m[id] = struct{}{}
+	}
+
+	res := make([]*ammodels.GettableAlert, 0, len(ids))
 	for _, a := range alerts {
-		if *a.Fingerprint == id {
-			return a, nil
+		if _, ok := m[*a.Fingerprint]; ok {
+			res = append(res, a)
 		}
 	}
 
-	return nil, status.Errorf(codes.NotFound, "Alert with id %s not found", id)
+	return res, nil
 }
 
-// Silence mutes alert with specified id.
-func (svc *Service) Silence(ctx context.Context, id string) error {
-	a, err := svc.FindAlertByID(ctx, id)
+// Silence mutes alerts with specified ids.
+func (svc *Service) Silence(ctx context.Context, ids []string) error {
+	if len(ids) == 0 {
+		return nil
+	}
+
+	alerts, err := svc.FindAlertsByID(ctx, ids)
 	if err != nil {
 		return err
 	}
 
-	if len(a.Status.SilencedBy) != 0 {
-		// already silenced
-		return nil
-	}
+	return silenceAlerts(ctx, alerts)
+}
 
-	matchers := make([]*ammodels.Matcher, 0, len(a.Labels))
-	for label, value := range a.Labels {
-		matchers = append(matchers,
-			&ammodels.Matcher{
-				IsRegex: pointer.ToBool(false),
-				Name:    pointer.ToString(label),
-				Value:   pointer.ToString(value),
-			})
+// SilenceAll mutes all available alerts.
+func (svc *Service) SilenceAll(ctx context.Context) error {
+	alerts, err := svc.GetAlerts(ctx)
+	if err != nil {
+		return err
 	}
 
-	starts := strfmt.DateTime(time.Now())
-	ends := strfmt.DateTime(time.Now().Add(100 * 365 * 24 * time.Hour)) // Mute for 100 years
-	_, err = amclient.Default.Silence.PostSilences(&silence.PostSilencesParams{
-		Silence: &ammodels.PostableSilence{
-			Silence: ammodels.Silence{
-				Comment:   pointer.ToString(""),
-				CreatedBy: pointer.ToString("PMM"),
-				StartsAt:  &starts,
-				EndsAt:    &ends,
-				Matchers:  matchers,
+	return silenceAlerts(ctx, alerts)
+}
+
+func silenceAlerts(ctx context.Context, alerts []*ammodels.GettableAlert) error {
+	var err error
+	for _, a := range alerts {
+		if len(a.Status.SilencedBy) != 0 {
+			// Skip already silenced alerts
```

Comment:
Possible race condition where we use stale data (alert has been updated since we read it)

Reply:
Not a big deal. We grabbing these alerts from alertmanager API, and we silence them with another API call, so we can't guarantee the atomicity of that operation. In the worst case user is supposed just to repeat the request.

Reply:
it's good to add to doc imho.